### PR TITLE
added _classes for knn classifiers

### PR DESCRIPTION
**Pull request by Hk669**
### **Added _classes attribute for k-Nearest Neighbors (KNN) Algorithm**

**Description:**
This pull request addresses the issue [#669 ](https://github.com/mljar/mljar-supervised/issues/669) and implements the addition of the _classes attribute for the k-Nearest Neighbors algorithm.

**Changes Made:**

Modified the KNNFit class to include the _classes attribute, providing unique classes based on the fitted model.
Updated the KNeighborsAlgorithm and KNeighborsRegressorAlgorithm classes to inherit from KNNFit and ClassifierMixin and RegressorMixin, respectively.
Adjusted relevant unit tests to ensure the proper functioning of the _classes attribute.

**Testing:**

- [x] Unit tests (test_knn.py) have been added/modified to check the functionality of the _classes attribute.
- [x] All tests pass successfully.

## checklist:
 - [x] Added _classes attribute.
 - [x] Updated unit tests.
 - [x] Checked compatibility and consistency.
tests_knn1
tests_knn
classes_property

### CODE REVIEW

File: supervised/algorithms/knn.py [hunk 1]
```diff
@@ -55,8 +55,16 @@ def fit(
...
+    def _classes(self):
+        # Returns the unique classes based on the fitted model
+        if hasattr(self.model, "classes_"):
+            return self.model.classes_
+        else:
+            return None
+
 
-class KNeighborsAlgorithm(KNNFit, RegressorMixin):
+class KNeighborsAlgorithm(KNNFit, ClassifierMixin):
```
Comment (pplonski) [inline supervised/algorithms/knn.py:67]: this looks strange, we derive from `ClassifierMixin` in KNNFit as well ...

File: supervised/algorithms/knn.py [hunk 2]
```diff
@@ -73,7 +81,7 @@ def __init__(self, params):
         )
 
 
-class KNeighborsRegressorAlgorithm(KNNFit):
+class KNeighborsRegressorAlgorithm(KNNFit, RegressorMixin):
```
Comment (pplonski) [inline supervised/algorithms/knn.py:84]: we derive from `KNNFit` which derives from `ClassifierMixin` and we also derive from `RegressorMixin`. This looks strange ...

File: tests/tests_algorithms/test_knn.py [hunk 3]
```diff
@@ -77,3 +81,20 @@ def test_is_fitted(self):
...
+        model.fit(self.X,self.y)
+
+
+        try:
+            classes = model._classes  
+        except AttributeError:
+            classes = None
+
+        #debug statements
+        print("np.unique(self.y):", np.unique(self.y))
```
Comment (pplonski) [inline tests/tests_algorithms/test_knn.py:97]: we dont need debug statements, let use asserts

**Review by pplonski — CHANGES_REQUESTED** [review]
Thank you for PR. Please check how classes are derived, because now we get some strange nesting between base classes.

**Comment by Hk669** [comment]
made the required changes, and its working quite great. 
passed all the unitttests.
tests_knn_new

**Review by pplonski — APPROVED** [review]

**Comment by pplonski** [comment]
Thank you @Hk669 for PR, good job :1st_place_medal:

**Comment by Hk669** [comment]
Thanks for the support @pplonski

**Pull request merged**